### ISSUE TITLE
Type checking silently skips bad typing after encountering module cattr

### ISSUE THREAD

**Issue by petergaultney** [python/mypy#6643]
I'm not sure whether this has anything to do with `cattrs` _specifically_, but it does seem to be related to the location of the usage of that module within the test file.  Any type error placed before its first usage is correctly caught, and any (that I've tried) placed after it is not correctly caught.

* Are you reporting a bug, or opening a feature request?

Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```
import typing as ty
import cattr


def drop_nonattrs(d: dict, Type: type) -> dict:
    """gets rid of all members of the dictionary that wouldn't fit in the given 'attrs' Type"""
    attrs_attrs = getattr(Type, '__attrs_attrs__', None)
    if attrs_attrs is None:
        raise ValueError(f'type {Type} is not an attrs class')
    attrs: ty.Set[str] = {attr.name for attr in attrs_attrs}

    combined_dict = dict()
    combined_dict += 3   # this line is an error and is caught

    return {key: val for key, val in d.items() if key in attrs}


def structure_ignore_extras(d: dict, Type: type):
    """This will drop top-level attributes that aren't expected"""
    combined_dict = dict()
    combined_dict += 3   # this line is an error and is caught

    return cattr.structure(drop_nonattrs(d, Type), Type)


def structure(*args, **kwargs):
    """A wrapper so that you only have to import this module"""
    combined_dict = dict()
    combined_dict += 3   # this line is an error and is NOT caught

    return cattr.structure(*args, **kwargs)


def combine(*args, **kwargs):
    """Allows you to combine multiple objects before structuring a new one!"""
    combined_dict = dict()
    combined_dict += 3   # this line is an error and is NOT caught
```

* What is the actual behavior/output?

```
$ mypy test.py --ignore-missing
test.py:12: error: Need type annotation for 'combined_dict'
test.py:13: error: Unsupported operand types for + ("Dict[Any, Any]" and "int")
test.py:13: error: Incompatible types in assignment (expression has type "int", variable has type "Dict[Any, Any]")
test.py:20: error: Need type annotation for 'combined_dict'
test.py:21: error: Unsupported operand types for + ("Dict[Any, Any]" and "int")
test.py:21: error: Incompatible types in assignment (expression has type "int", variable has type "Dict[Any, Any]")
```

* What is the behavior/output you expect?

I would expect similar errors for lines 28, 29, 36, and 37.

* What are the versions of mypy and Python you are using?

0.700, 3.7.2

  Do you see the same issue after installing mypy from Git master?

Yes

* What are the mypy flags you are using? (For example --strict-optional)

`--ignore-missing-imports`, but it happens with or without it.

**Comment by ilevkivskyi** [comment]
Non-annotated functions are not checked, use `--check-untyped-defs` if you want them to be checked anyway.

**Comment by petergaultney** [comment]
Wow, I've been using MyPy for a year and didn't realize it wasn't checking those at all.  I see it in the documentation now, but I guess I hadn't ever really read the intro bits.  

Kind of terrifying, to be honest, but good to know.  Thanks.
